src/main/java/com/redhat/cloud/notifications/migration/policynotification/PoliciesMigrationService.java

```diff
@@ -0,0 +1,103 @@
+import com.redhat.cloud.notifications.models.EmailSubscriptionType;
+import com.redhat.cloud.notifications.models.Endpoint;
+import com.redhat.cloud.notifications.models.EndpointType;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.Uni;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+import javax.inject.Inject;
+import javax.ws.rs.Consumes;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import java.util.HashSet;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Path("/internal/policies-notifications")
+@Consumes(MediaType.APPLICATION_JSON)
+@Produces(MediaType.APPLICATION_JSON)
+public class PoliciesMigrationService {
+
+    static final String BUNDLE = "insights";
+    static final String APPLICATION = "policies";
+    static final String EVENT_TYPE = "policy-triggered";
+    static final String DAILY_EMAIL_TYPE = "policies-daily-mail";
+    static final String INSTANT_EMAIL_TYPE = "policies-instant-mail";
+
+    @Inject
+    @RestClient
+    PolicyNotifications policyNotifications;
+
+    @Inject
+    EndpointEmailSubscriptionResources emailSubscriptionResources;
+
+    @Inject
+    EndpointResources endpointResources;
+
+    @Inject
+    ApplicationResources applicationResources;
+
+    static class MigrateResponse {
+        public AtomicInteger eventTypesMigrated = new AtomicInteger();
+        public AtomicInteger accountsMigrated = new AtomicInteger();
+    }
+
+    @Path("/migrate")
+    @GET
+    public Uni<MigrateResponse> migrate() {
+        final MigrateResponse response = new MigrateResponse();
+        return applicationResources.getEventType(BUNDLE, APPLICATION, EVENT_TYPE)
+        .onItem().transformToMulti(eventType -> {
+            return policyNotifications.getSubscriptions()
+                    .onItem().transformToMulti(policiesEmailSubscriptions -> Multi.createFrom().iterable(policiesEmailSubscriptions))
+                    .onItem().transformToMulti(pes -> emailSubscriptionResources.subscribe(
+                    pes.accountId,
+                    pes.userId,
+                    BUNDLE,
+                    APPLICATION,
+                    fromPoliciesEventType(pes.eventType.toLowerCase())
+            )
+                    .invoke(() -> response.eventTypesMigrated.incrementAndGet())
+                    .onItem().transform(subscribed -> pes.accountId).toMulti())
+                    .concatenate().collectItems().in(HashSet::new, HashSet::add)
+                    .onItem().transformToMulti(accountIds -> Multi.createFrom().iterable(accountIds))
+                    .onItem().castTo(String.class)
+                    .onItem().transformToMultiAndConcatenate(accountId -> {
+                        response.accountsMigrated.incrementAndGet();
+                        // Remove existing endpoints
+                        return endpointResources.getLinkedEndpoints(accountId, eventType.getId(), new Query())
```

Comment:
Is there a chance this could return more than one linked endpoint per `accountId`?

Reply:
Yes, it could return multiple.
I'm returning a Multi here, I'm assuming that all the linked endpoints will get unlink in the next step, later they will get collected and the result not really used.

Reply:
Yeah I saw the result was ignored below, but if `n` endpoints are unliked here, doesn't it mean that we try to create `n` EMAIL_SUBSCRIPTION endpoints for a single `accountId/eventType` tuple?

Reply:
Oh no, it's all transformed into a List before creating the endpoint, I missed that :)

Reply:
I'd rather ask stupid question rather than miss something 😄

Reply:
I'm rather new to this reactive/mutiny stuff, thanks for double checking it!